packages/hebao_v2/contracts/lib/SignatureUtil.sol

```diff
@@ -115,31 +115,29 @@ library SignatureUtil
             return false;
         }
 
-        uint signatureTypeOffset = signature.length.sub(1);
-        SignatureType signatureType = SignatureType(signature.toUint8(signatureTypeOffset));
+	require(signature.length == 65 || signature.length == 66, "INVALID_SIGNATURE_LENGTH");
 
-        // Strip off the last byte of the signature by updating the length
-        assembly {
-            mstore(signature, signatureTypeOffset)
-        }
+	if (signature.length == 66) {
+	    // Strip off the last byte of the signature by updating the length
+	    assembly {
+		mstore(signature, 65)
+	    }
+	}
 
-        if (signatureType == SignatureType.EIP_712) {
-            success = (signer == recoverECDSASigner(signHash, signature));
-        } else if (signatureType == SignatureType.ETH_SIGN) {
+	success = (signer == recoverECDSASigner(signHash, signature));
```

Comment:
This does increase the gas cost of verifying `ETH_SIGN` signatures by ~3000 gas if each time it needs to be checked if it's not a valid `EIP_712` signature. I guess not too bad if most times signatures are `EIP_712`.